octodns/record/ds.py

```diff
@@ -70,28 +70,20 @@ def process(cls, values):
     def __init__(self, value):
         super().__init__(
             {
-                'flags': int(value['flags']),
-                'protocol': int(value['protocol']),
+                'key_tag': int(value['key_tag']),
```

Comment:
These will need something like:

```suggestion
                'key_tag': int(value.get('flags', None) or value['key_tag']),
```

Reply:
I did the same here, it is a bit cumbersome, but I personally prefer it since it's an either or situation where I don't we want anyone mixing the fields. Also again with algorithm being present both field sets makes it difficult to handle.